agent/util-scripts/require-rpm

```diff
@@ -18,47 +18,49 @@ else
     _match="eq"
 fi
 
-_status=$(rpm --query --queryformat="%{EVR}" ${_rpm} 2>/dev/null)
-rc=0
-if echo ${_status} | grep -q "is not installed"; then
-    rc=1
+# The inputs for the `rpmdev-vercmp` tool to compare are EVR tags.  The first
+# tag is derived from a look up of the RPMs that are installed.  Note well that
+# this can be a list of multiple RPMs in some cases.  An "EVR" stands for
+# "epoch", "version", and "release".
+_evr_list=$(rpm --query --queryformat="%{EVR}" ${_rpm} 2>/dev/null)
+rc=1
+if echo ${evr_list} | grep -q "is not installed"; then
+    echo "${_rpm}${_version}"
 else
-    # Some RPMs might have multiple version installed, iterate through all of
+    # Some RPMs might have multiple versions installed, iterate through all of
     # them.
-    for rpm_installed in ${_status}; do
-		# Exit status is 0 if the EVR's are equal, 11 if EVR1 is newer, and 12
+    for rpm_installed in ${_evr_list}; do
+        # Exit status is 0 if the EVR's are equal, 11 if EVR1 is newer, and 12
         # if EVR2 is newer.  Other exit statuses indicate problems.
-        rpmdev-vercmp ${rpm_installed} ${_rpm}${_version} > /dev/null 2>&1
+        _stderr="$(rpmdev-vercmp ${rpm_installed} ${_rpm}${_version} 2>&1 > /dev/null)"
```

Comment:
I don't think you need the quotation marks -- I think the assignment gets the full string without them -- and having them just makes things more complicated if we want to add quotes inside them (as we saw elsewhere...).

Reply:
Dropped.